tests/old_functional/test_direct_transfer.py

```diff
@@ -838,6 +838,155 @@ def test_sessions(self):
                     matches = list(filter(expression.match, records))
                     assert len(matches) == x + 1
 
+    def test_pause_resume_session(self):
+        """
+        Test the session pause and resume system.
+        The Session final status should be COMPLETED.
+        """
+
```

Comment:
You should define that here and use it in the whole test.
```suggestion
    engine = self.engine_1

```